doc/src/manual/interfaces.md

```diff
@@ -322,8 +344,8 @@ forms, in most case you only need to specialize the 3-argument form.) For this t
 that `SparseArray` is mutable (supports `setindex!`). [`similar()`](@ref) is also used to allocate result
 arrays for arithmetic on `AbstractArrays`, for instance:
 
-```julia
-julia> A + 4
+```jldoctest squarevectype
+julia> A .+= 4
```

Comment:
No straightforward, safe way of specifying the output type of `broadcast` over combinations involving a specified user-defined type exists at present; this issue is one of the wrinkles we need iron out when redesigning `Broadcast`'s container type promotion mechanism.

Calling `broadcast!` as above is the simplest workaround. Otherwise you must define a set of `promote_containertype` methods, a `broadcast_c` method, and sometimes a `broadcast_indices` method, with the `promote_containertype` methods funneling relevant combinations of `broadcast` arguments to the `broadcast_c` method.